products/bridge/smart-contracts/contracts/core/ShardDispatcher.sol

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity ^0.8.20;
+
+import {Relayer} from "contracts/core/Relayer.sol";
+import {DispatchReplayChecker} from "contracts/core/DispatchReplayChecker.sol";
+
+interface IShardDispatcherErrors {
+    error Unauthorized();
+}
+
+interface IShardDispatcherEvents {
+    event Dispatched(uint indexed sourceShardId, uint indexed nonce);
```

Comment:
@theo-zil is this satisfactory amount of events dispatched? Would you like me to add more events? I guess the only purpose of this is for replay protection right?

Reply:
I think this won't even be used actually, but won't hurt. Can be expanded (or removed) later as needed

Reply:
The only scenario I could imagine to be useful is to check if a certain event has already been dispatched. On one hand the validator could just do a simulated call to the RPC, but they can't verify a simulated call. So perhaps, having a way to know that an execution has failed is necessary

Reply:
The `Dispatched` event was designed to be used only in the cross-chain case, where it is essential for the validators running light clients to know if one of them has already submitted the transaction for a pending `nonce`. In the cross-shard case, we could omit the event, or keep it as an information for light clients on the source shard to know if an cross-shard call has been executed.

Reply:
@theo-zil happy for you to make the call. I could remove it on a follow-up PR if completely unnecessary